lib/rubocop/cop/util.rb

```diff
@@ -212,32 +212,32 @@ def double_quotes_required?(string)
 
         # Regex matches IF there is a ' or there is a \\ in the string that is
         # not preceded/followed by another \\ (e.g. "\\x34") but not "\\\\".
-        string.inspect =~ /'|(?<! \\) \\{2}* \\ (?![\\"])/x
+        string =~ /'|(?<! \\) \\{2}* \\ (?![\\"])/x
       end
 
       # If double quoted string literals are found in Ruby code, and they are
       # not the preferred style, should they be flagged?
       def double_quotes_acceptable?(string)
         # If a string literal contains hard-to-type characters which would
         # not appear on a "normal" keyboard, then double-quotes are acceptable
-        double_quotes_required?(string) ||
+        needs_escaping?(string) ||
           string.codepoints.any? { |cp| cp < 32 || cp > 126 }
       end
 
+      def needs_escaping?(string)
+        double_quotes_required?(string.inspect)
```

Comment:
Hi @jonas054, thanks for reviewing this.

If you see 5b84608c, this is just a refactor to extract the `string.inspect` call to a helper method. Since some usages of `double_quotes_required?` were assigning to an `escape` variable I figured `needs_escaping?` could be a good name for this new auxiliary method.

The reason for this refactor is that in https://github.com/bbatsov/rubocop/pull/3189/commits/e64d0bb99bc77cb049dffe6a4c086e3ceb1c5cf8 I need to call `double_quotes_required?` on something that's already inspected. The reason for that can be seen in this debugging session:

```
deivid@pantani ~/Code/rubocop $ bundle exec rspec spec/rubocop/cop/style/string_literals_spec.rb:102
Run options:
  include {:focus=>true, :locations=>{"./spec/rubocop/cop/style/string_literals_spec.rb"=>[102]}}
  exclude {:broken=>#<Proc:./spec/spec_helper.rb:30>}

Randomized with seed 4386

[7, 16] in /home/deivid/Code/rubocop/lib/rubocop/cop/mixin/string_literals_help.rb
    7:     module StringLiteralsHelp
    8:       include StringHelp
    9: 
   10:       def wrong_quotes?(node)
   11:         require 'byebug'; byebug
=> 12:         src = node.source
   13:         return false if src.start_with?('%', '?')
   14:         if style == :single_quotes
   15:           !double_quotes_required?(src)
   16:         else
(byebug) node.source
"\"Espa\\u00f1a\""
(byebug) node.str_content
"España"
(byebug)
```

`node.source` keeps the unicode control characters as they were originally, so that's what's needed for fixing these unicode issues. But as you can see, that string is already inspected, so I need to perform the regex matching done in `double_quotes_required?` _without_ inspecting the string first. That's why I extracted the `needs_escaping?` method and moved inspecting the string out of `double_quotes_required?`.

I hope this makes sense and helps clarifying! Based on the explanation, what can I do to make the patch clearer? I'm thinking maybe moving `string.inspect` to a `unescape` method could help making the code more self-explanatory?